internal/agent/install.go

```diff
@@ -110,14 +133,24 @@ func Install(dir ...string) error {
 		fmt.Printf("- config not found in the system: %s", err.Error())
 	}
 
-	_, err = bus.Manager.Publish(events.EventChallenge, events.EventPayload{Config: cc.String()})
+	agentConfig, err := LoadConfig()
 	if err != nil {
 		return err
 	}
 
+	// try to clear screen
+	fmt.Print("\033c")
+
 	cmd.PrintBranding(DefaultBanner)
 
-	agentConfig, err := LoadConfig()
+	// If there are no providers registered, we enter a shell for manual installation and print information about
+	// the webUI
+	if len(bus.Manager.Plugins) == 0 {
```

Comment:
I'd recommend to start using predicate functions for readability and flexibility for future changes. For example something like `bus.Manager.HasRegisteredProviders()`

Reply:
good catch! addressed in https://github.com/kairos-io/kairos/pull/620/commits/8ae581024504ea5a84c730c547cbaa2c22e01d17 !